dot/state/initialize.go

```diff
@@ -40,8 +40,11 @@ func (s *Service) Initialise(gen *genesis.Genesis, header *types.Header, t *trie
 
 	s.db = db
 
-	if err = t.WriteDirty(database.NewTable(db, storagePrefix)); err != nil {
-		return fmt.Errorf("failed to write genesis trie to database: %w", err)
+	// TODO: all trie related db operations should be done in pkg/trie
```

Comment:
Should we have an issue to represent this?

Reply:
That's a good question, I don't think so since we are going to replace it with lazy loading and that implementation encapsulates all the DB related operations in the same trie. It will execute the writes before calculating the root hash, so I don't think we need an issue for this, it's more an explanation for the casting 😅